test/mocking_classes.py

```diff
@@ -126,6 +126,139 @@ def rand_step(self, tensordict: Optional[TensorDictBase] = None) -> TensorDictBa
         return self.step(tensordict)
 
 
+class MockBatchedLockedEnv(EnvBase):
+    def __init__(self, device):
+        super(MockBatchedLockedEnv, self).__init__(device=device)
+        self.action_spec = NdUnboundedContinuousTensorSpec((1,))
+        self.input_spec = CompositeSpec(
+            action=NdUnboundedContinuousTensorSpec((1,)),
+            observation=NdUnboundedContinuousTensorSpec((1,)),
+        )
+        self.observation_spec = CompositeSpec(
+            next_observation=NdUnboundedContinuousTensorSpec((1,))
+        )
+        self.reward_spec = NdUnboundedContinuousTensorSpec((1,))
+        self.is_closed = False
+        self.counter = 0
+
+    @classmethod
+    def __new__(cls, *args, **kwargs):
+        return super().__new__(cls, *args, _batch_locked=True, **kwargs)
+
+    def set_seed(self, seed: int) -> int:
+        assert seed >= 1
+        self.seed = seed
+        self.counter = seed % 17  # make counter a small number
+        self.max_val = max(self.counter + 100, self.counter * 2)
+        return seed_generator(seed)
+
+    def _step(self, tensordict):
+        self.counter += 1
+        n = (
+            torch.full(tensordict.batch_size, self.counter)
+            .to(self.device)
+            .to(torch.get_default_dtype())
+        )
+        done = self.counter >= self.max_val
+        done = torch.full(
+            tensordict.batch_size, done, dtype=torch.bool, device=self.device
+        )
+
+        return TensorDict(
+            {"reward": n, "done": done, "next_observation": n}, tensordict.batch_size
+        )
+
+    def _reset(self, tensordict: TensorDictBase, **kwargs) -> TensorDictBase:
+        self.max_val = max(self.counter + 100, self.counter * 2)
+
+        if tensordict is None:
+            batch_size = self.batch_size
+        else:
+            batch_size = tensordict.batch_size
+
+        n = (
+            torch.full(batch_size, self.counter)
+            .to(self.device)
+            .to(torch.get_default_dtype())
+        )
+        done = self.counter >= self.max_val
+        done = torch.full(batch_size, done, dtype=torch.bool, device=self.device)
+
+        return TensorDict(
+            {"reward": n, "done": done, "next_observation": n}, batch_size
+        )
+
+    def rand_step(self, tensordict: Optional[TensorDictBase] = None) -> TensorDictBase:
+        return self.step(tensordict)
+
+
+class MockBatchedUnLockedEnv(EnvBase):
```

Comment:
does this inherit from another env? Can we inherit things like set_seed etc?
It feel that only the `__new__` method changes, am I right?
Perhaps we could just inherit from the other mocking class and change the `__new__` for he non-batch_locked version?